### DISCUSSION TITLE
NIBR substructure filters acetal SMARTS issue

### DISCUSSION THREAD

**Discussion by marco-chimfarm** [rdkit/rdkit#7875] [Q&A]
Hi everyone,

I have noticed that within the [NIBR filters](https://github.com/rdkit/rdkit/blob/master/Contrib/NIBRSubstructureFilters/SubstructureFilter_HitTriaging_wPubChemExamples.csv) there is this SMARTS pattern: "acetal_aminal_acyclic" (line 20): 
`[O,S&v2,N]-!@[#6;$(C=O),X4]-!@[O,S&v2,N]`

This SMARTS pattern is flagging ureas and carbonates as acetals, since it forces the central carbon atom to be bound to a carbonyl.
Do you know if this is an intended behaviour or not?
To me it looks like the correct SMARTS pattern would be this one:
`[O,S&v2,N]-!@[#6;!$(C=O),X4]-!@[O,S&v2,N]`
With the carbonyl being explicitely excluded from the match of the central carbon atom to only flag proper acetals/aminals.

Kind regards

**Comment by greglandrum** [comment]
Hi @marco-chimfarm that's a great question. @NadineSchneider this was your work, do you have any thoughts?

**Comment by marco-chimfarm** [comment]
Hi @greglandrum, thanks for checking on this issue! Do you have any news about it?

**Comment by JanoschMenke** [comment]
I might be a little late to the party, but @marco-chimfarm change to the SMARTS pattern should be correct. Otherwise, the pattern will always flag substructures such as urea which we would not want to flag
